tests/common/plugins/sanity_check/__init__.py

```diff
@@ -287,10 +287,15 @@ def recover_on_sanity_check_failure(duthosts, failed_results, fanouthosts, local
                     infra_recovery_actions.append(failed_result['action'])
         for action in infra_recovery_actions:
             action()
-        for dut_name, dut_results in list(dut_failed_results.items()):
-            # Attempt to restore DUT state
-            recover(duthosts[dut_name], localhost, fanouthosts, nbrhosts, tbinfo, dut_results,
-                    recover_method)
+
+        is_modular_chassis = duthosts.nodes[0].get_facts().get("modular_chassis")
+        if is_modular_chassis:
+            recover_chassis(duthosts)
```

Comment:
Is it safe to just roll out this new way of recovering to all chassis, or do we need to add a condition say only when it's Cisco-8000?

Reply:
Yes, it's safe
it's safe to do config reload on all the RP/LCs
The only risk is we don't use reboot during the recover, but I believe we need to consider more carefully for rebooting to recover the chassis, we can support it in the future.
So the change is supposed to help rather than defect other chassis.